cloud/aws/bin/dumpdb.py

```diff
@@ -0,0 +1,142 @@
+        "\n")
+    # Current working dir should be the 'checkout' folder, so go one level above.
+    default_file = os.path.join(
+        os.path.dirname(os.getcwd()), 'civiform_database.dump')
+    dumpfile = input(
+        f"Enter the location to save the dump file (default: {default_file}): "
+    ) or default_file
+
+    dumpdir = os.path.dirname(dumpfile)
+    if not os.path.isdir(dumpdir):
+        os.makedirs(dumpdir)
+        print(f'Directory created: {dumpdir}')
+
+    # Generate a new key pair for the dbaccess instance. We'll
+    # save this to a temp directory and run all the critical pieces
+    # inside this block so we ensure the key is cleaned up.
+    with tempfile.TemporaryDirectory() as tmpdir:
+        print(f'Generating key pair in {tmpdir}')
+        _run_cmd(f'ssh-keygen -t rsa -b 4096 -f {tmpdir}/dbaccess -N ""')
+
+        print('Deploying dbaccess instance')
+        os.environ[
+            'TF_VAR_dbaccess_cidr_allowlist'] = f'["{_detect_public_ip()}/32"]'
+        os.environ['TF_VAR_dbaccess'] = "true"
+        os.environ['TF_VAR_dbaccess_public_key'] = f'{tmpdir}/dbaccess.pub'
+
+        try:
+            _run_terraform(config)
+
+            ec2_host_ip = aws.get_dbaccess_ec2_host_ip()
+            print(f'EC2 host IP is {ec2_host_ip}')
+
+            db_hostname = aws.get_database_hostname()
+            db_user = aws.get_secret_value(
+                config.app_prefix + '-civiform_postgres_username')
+            db_pwd = aws.get_secret_value(
+                config.app_prefix + '-civiform_postgres_password')
+
+            args = f'-o "UserKnownHostsFile=/dev/null" -o "StrictHostKeyChecking=no" -o "IdentitiesOnly=yes" -i {tmpdir}/dbaccess'
+            ssh = f'ssh -q {args} "ubuntu@{ec2_host_ip}" '
+
+            # We typically need about 15 seconds before the EC2 instance is ready
+            # to accept SSH connections.
+            print('Waiting for SSH access to EC2 host to become available')
+            sleep(15)
+            while True:
+                try:
+                    _run_cmd(ssh + 'exit', quiet=True)
+                    break
+                except subprocess.CalledProcessError as e:
+                    if e.returncode == 255:
+                        print(
+                            'SSH connection failed. Retrying in 10 seconds...')
+                        sleep(10)
+                    else:
+                        raise e
+
+            print('Installing postgresql-client')
+            cmd = ssh + '"sudo apt-get update && sudo apt-get install -y postgresql-client"'
+            _run_cmd(cmd)
+
+            print('Generating dump file')
+            cmd = ssh + f"PGPASSWORD='{db_pwd}' pg_dump -w -Fc -h '{db_hostname}' -U '{db_user}' -d postgres > civiform_database.dump"
+            _run_cmd(cmd)
+
+            print('Downloading dump file to local machine')
```

Comment:
**suggestion:** I assume you'll have to run this command from the terminal on your computer (or a vm or something) and can't run it on cloudshell or something because then the file would be kind of stuck in that cloudshell instance related to that particular AWS deployment? If that's the case, I think it could be worth including a separate prompt after the warning ensuring that the user is running this from the right place. We may also want to check where the deployers of cf today run their deployment from to confirm this would work for them.

Reply:
Makes sense - probably worth understanding where each entity runs their deployment from so we can make sure we support all the methods.